src/main/pages/che-7/end-user-guide/ref_devfile-reference.adoc

```diff
@@ -3,58 +3,67 @@
 
 This section contains devfile reference and instructions on how to use the various elements that devfiles consist of.
 
+
 == Adding projects to a devfile
 
-In most cases devfile contains one or more projects for which development the whole workspace is created.
-Projects are added in `projects` section of devfile.
+In most cases, a devfile contains one or more projects. A workspace is created to develop those projects. Projects are added in the `projects` section of devfiles.
+
+Each project in a single devfile must have:
 
-Each project in a single devfile must have a unique name.
+* Unique name
+* Source specified
 
-Beside project name each project must have its source specified.
 Project source consists of two mandatory values: `type` and `location`.
 
-`type` defined kind of project source provider.
- +
-`location` contains URI to where project source is located.
+`type`:: The kind of project-source provider.
+`location`:: The URL of project source.
+
+Che supports the following project types:
+
+`git`:: Projects with sources in Git. The location points to a clone link.
 
-Che support the following project types:
+`github`:: Same as `git` but for projects hosted on link:https://github.com/[GitHub] only. Use `git` for projects that do not use GitHub-specific features.
 
- * `git` - projects which is under git VCS. Location should point to clone link.
- * `github` - the same as above, but for projects hosted on https://github.com/[GitHub] only.
-   If there is no plan to use GitHub specific features, one may use `git` type instead.
- * `zip` - projects which sources is in zip archive. Location should point to zip file.
+`zip`:: Projects with sources in a ZIP archive. Location points to a ZIP file.
 
-=== The full project source specification of type `git`:
+
+=== Project-source type: git
 
 [source,yaml]
 ----
 source:
     type: git
     location: https://github.com/eclipse/che.git
-    startPoint: master
+    startPoint: master           <1>
     tag: 7.2.0
     commitId: 36fe587
     branch: master
-    sparseCheckoutDir: wsmaster
+    sparseCheckoutDir: wsmaster  <2>
 ----
+<1> `startPoint` is the general value for `tag`, `commitId`, and `branch`. The `startPoint`, `tag`, `commitId`, and `branch` parameters are mutually exclusive. When more than one is supplied, the following order is used: `startPoint`, `tag`, `commitId`, `branch`.
+<2> `sparseCheckoutDir` the template for the sparse checkout Git feature.
 
-`startPoint` is general value for `tag`, `commitId` and `branch`.
-Note, that `startPoint`, `tag`, `commitId` and `branch` are mutually exclusive.
-However, if a few of them are given the following order is used: `startPoint`, `tag`, `commitId`, `branch`.
+////
+TODO: @mmorhun -- I don't understand this sentence. Could you please try to reword it?
```

Comment:
I wanted to say, that sparse checkout feature might be useful when a user wants to have cloned only a part (in most cases it is a single directory) of a project.
The sentence above is designed to briefly explain what sparse checkout does (for users who's not familiar with that git feature). If you think it is redundant - feel free to remove it.

Reply:
Got it. I added it to the explanation.